pkg/utils/metrics.go

```diff
@@ -45,21 +46,126 @@ func NewAgentMetrics(ai *AgentInfo) AgentMetrics {
 		Help:        "Total number of reports (keepalive messages) from the agent.",
 	})
 
-	if counter, ok := tryRegister(am.ErrorCount); !ok {
+	if counter, ok := tryRegisterCounter(am.ErrorCount); !ok {
 		// use existing counter
 		am.ErrorCount = counter
 	}
-	if counter, ok := tryRegister(am.ReportCount); !ok {
+	if counter, ok := tryRegisterCounter(am.ReportCount); !ok {
 		// use existing counter
 		am.ReportCount = counter
 	}
 
+  am.ProbeConnectionResult = prometheus.NewGaugeVec(
```

Comment:
Every GaugeVec has uniq name and description. Not sure that iterating over the loop will make it a lot better since we'll have to declare a hash/array of names/descriptions to iterate over. If I understood you correctly :)